transports/janus_websockets.c

```diff
@@ -776,7 +774,8 @@ static void janus_websockets_destroy_client(
 	ws_client->buflen = 0;
 	ws_client->bufpending = 0;
 	ws_client->bufoffset = 0;
-	janus_mutex_unlock(&ws_client->mutex);
+	janus_mutex_unlock(&ws_client->ts->mutex);
+	janus_transport_session_destroy(ws_client->ts);
```

Comment:
This has a tricky dependency.  Libwebsockets cannot free `ws_client` until `janus_websockets_destroy_client` returns.  By calling `janus_transport_session_destroy` last, we make sure that `ws_client->ts` doesn't get freed earlier by `janus_refcount_decrease`.  Anybody else acquiring `ws_client->ts->mutex` will find that `ws_client->ts->transport_p` is now `NULL`.  However, if they didn't increase the refcount, they may access `ws_client->ts->transport_p` via a freed pointer.

One could instead use the following sequence (but this still doesn't protect somebody waiting on the mutex from having it freed under them).  This twists my brain.

	janus_refcount_increase(&ws_client->ts-ref);
	janus_transport_session_destroy(ws_client->ts);
	janus_mutex_unlock(&ws_client->ts->mutex);
	janus_refcount_decrease(&ws_client->ts-ref);